dev/test/recursive-delete.ts

```diff
@@ -235,8 +239,30 @@ describe('recursiveDelete() method:', () => {
     });
 
     it('creates a second query with the correct startAfter', async () => {
-      const firstStream = Array.from(Array(MAX_PENDING_OPS).keys()).map(
-        (_, i) => result('doc' + i)
+      // This test checks that the second query is created with the correct
+      // startAfter() once the RecursiveDelete instance is below the
+      // MIN_PENDING_OPS threshold to send the next batch. Use lower limits
+      // than the actual RecursiveDelete class in order to make this test run fast.
+      const maxPendingOps = 100;
+      const minPendingOps = 11;
+      const maxBatchSize = 10;
+      const cutoff = maxPendingOps - minPendingOps;
+      let numDeletesBuffered = 0;
+
+      // This deferred promise is used to delay the BatchWriteResponses from
+      // returning in order to create the situation where the number of pending
+      // operations is less than `minPendingOps`.
+      const bufferDeferred = new Deferred<void>();
+
+      // This deferred completes when the second query is run.
+      const secondQueryDeferred = new Deferred<void>();
+
+      const firstStream = Array.from(Array(maxPendingOps).keys()).map((_, i) =>
```

Comment:
Can you create a helper variable for `Array.from(Array(maxPendingOps).keys())` and name it appropriately? It is used twice and not very pretty :)

Reply:
Done.